clients/ui/frontend/src/app/pages/modelCatalog/screens/ModelDetailsPage.tsx

```diff
@@ -7,36 +7,30 @@ import {
   ContentVariants,
   Flex,
   FlexItem,
-  Label,
   Skeleton,
   Stack,
   StackItem,
 } from '@patternfly/react-core';
-import { TagIcon } from '@patternfly/react-icons';
 import { ApplicationsPage } from 'mod-arch-shared';
-import { useModelCatalogSources } from '~/app/hooks/modelCatalog/useModelCatalogSources';
-import { extractVersionTag } from '~/app/pages/modelCatalog/utils/modelCatalogUtils';
+import { getModelName } from '~/app/pages/modelCatalog/utils/modelCatalogUtils';
 import ModelDetailsView from '~/app/pages/modelCatalog/screens/ModelDetailsView';
 
+import { useCatalogModel } from '~/app/hooks/modelCatalog/useCatalogModel';
+
 type RouteParams = {
-  modelId: string;
+  sourceId: string;
+  modelName: string;
+  repositoryName: string;
 };
 
 const ModelDetailsRoute: React.FC = () => {
-  const { modelId } = useParams<RouteParams>();
-  const { sources, loading, error } = useModelCatalogSources();
+  const { sourceId, repositoryName, modelName } = useParams<RouteParams>();
 
-  const model = React.useMemo(() => {
-    for (const source of sources) {
-      const found = source.models?.find((m) => m.id === modelId);
-      if (found) {
-        return found;
-      }
-    }
-    return undefined;
-  }, [sources, modelId]);
+  const state = useCatalogModel(sourceId || '', `${repositoryName}/${modelName}` || '');
+  const [model, modelLoaded, modelLoadError] = state;
 
-  const versionTag = extractVersionTag(model?.tags);
+  // TODO: we don't have tags prop on models
+  // const versionTag = extractVersionTag(model?.tags);
```

Comment:
currently we do not have tags property on CatalogModel, started a thread to followup on this. will update this based on this and other commented code in this PR for version/tags based on the conversation

Reply:
updated the code, we will most likely will not have version for models